lib/Sema/CSGen.cpp

```diff
@@ -3125,18 +3125,9 @@ namespace {
         auto expansionType =
             CS.getType(packEnvironment)->castTo<PackExpansionType>();
         CS.addConstraint(ConstraintKind::ShapeOf, expansionType->getCountType(),
-                         elementType,
+                         packType,
                          CS.getConstraintLocator(packEnvironment,
                                                  ConstraintLocator::PackShape));
```

Comment:
Interesting, this was completely backwards but worked... I wonder whether there are some checks we could add to archetype handling or shape-of constraint handling to fail if code like this is added in the future?